### DISCUSSION TITLE
Ideas for development

### DISCUSSION THREAD

**Discussion by ka-sarthak** [FAIRmat-NFDI/nomad-analysis#13] [Ideas]
Let's use this discussion to add ideas for improvement of `nomad-analysis`.  I will suggest adding a small description of the idea here and linking it with its issue. 

1. **Define general base sections for analysis:** #11 
    Parent sections like SpectralAnalysis(Analysis) can be inherited by technique-specific classes like XRDAnalysis. These parent classes will link methods from different domains (like computational and characterization) that use similar mathematical constructs to conduct analysis. The common methods defined by these parents allow for interoperability in the analysis plugin.

2. **Define static methods in base sections:** #15 
    This allows to have certain methods encapsulated within the base section, while also being available to be used as a standalone function. For example, having a static method `find_peaks` in `SpectralAnalysis` class can be used to automate the process of peak finding by using it in the `normalize` method. In addition, one can also use the method independently by calling `nomad_analysis.SpectralAnalysis.find_peaks`

3. **Automation in analysis classes:** #16 
    For the technique-specific classes, extend methods from their parents or define new ones to analyze the linked inputs. These methods can be used in the `normalize` method for automation of analysis workflow.

4.  **ELN classes handle JupteryNotebooks and make it easy to connect inputs:**
    The ELN classes shouldn't be focused on the automation of analysis (this should happen in technique-specific analysis classes). Instead, these classes should use the analysis classes and extend them with user-friendly GUI features. Two such features are: hosting the analysis code in the Jupyter Notebooks and simplifying linking of the input references. 
    - _Jupyter Notebooks:_ #17 
        Brings out the analysis code that is being used by the analysis class and allows for modification by the user. If the code is modified, the version on the notebooks should be persisted and used for generating results.
    - _Input references:_ #18 
        A new EditQuantity should be developed that combines the NOMAD search/explore to support the query of the inputs and generate multiple input references based on the search results. 

5. **Extend the outputs sub-section of the Analysis base section:** #19
    The results/outputs of the analysis should be stored in the outputs sub-section. This needs to be extended for each technique-specific analysis class. Extend it for XRD.

**Comment by JosePizarro3** [comment]
Everything looks good to me (note the heart 😄 ), thank you very much for opening this Discussion 👏🏻 

Next week I have a sprint for defining `Outputs` in the simulation schema. I will then come back to this the week after next one, with fresh ideas from this sprint.

I will then answer your points in the issues, and work on #11. Maybe then we will understand each other better, specially when talking about referencing.

**Reply by ka-sarthak** [reply]
Sounds good to me :)
